### ISSUE TITLE
Lint/UnusedMethodArgument should add an offence for overriding assignment

### ISSUE THREAD

**Issue by pocke** [bbatsov/rubocop#4823]
Lint/UnusedMethodArgument does not add offence for overriding assignment.

For example:

```ruby
# `x` is not used, but the cop does not add offence.
def foo(x)
  x = 1
  puts foo
end
```

Real world problem: https://github.com/bbatsov/rubocop/pull/4822

--------

## Expected behavior

The cop add an offence for the code.

## Actual behavior

The cop allows this code.

## Steps to reproduce the problem

Run rubocop with the code.

## RuboCop version

master

```
0.50.0 (using Parser 2.4.0.0, running on ruby 2.4.2 x86_64-linux)
```

## Note

`UnusedBlockArgument` has same problem.
But `Lint/UselessAssignment` works well.

**Comment by akhramov** [comment]
Huh... I think now it's my duty to fix this one 😃

**Comment by mvz** [comment]
@pocke I'm getting this offense now but I'm not sure how you want it to be fixed? The assignment has an effect on the `super` call, so removing it is a (breaking) change.

**Comment by akhramov** [comment]
@mvz, your case deserves a separate issue. I have created one (#4892). Thanks!
